### ISSUE TITLE
Packages to install have invalid keys or no keys. (Bugzilla Bug 284)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/test2#284]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 284
Date: 2013-11-08 15:53:14 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;
To:   OpenMandriva QA &lt;<bugs@openmandriva.org>&gt;
CC:   @benbullard79, @berolinux, @itchka, @tpgxyz

Duplicates:   #344
Depends on:   #344
Last updated: 2014-02-25 09:25:22 +0000

## Comment 1981
Date: 2013-11-08 15:53:14 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

Some packages give a warning at the time of install them because of an invalid or a missing key ID.

Examples:
flash-player-plugin-11.2.202.310-2 gives a "Medium without key"
unrar-4.20-3 gives a "invalid key ID".

This kind of problem was much more "massive" some months ago, but now it seems mostly solved, except for these cases (maybe others too, but I don't detect them).

I guess that the flash-player case is because the plugin is probably downloaded form an external site, not from a repository. If this is the case, then other packages, like skype for example, are prone to have the "medium without key" warning.

## Comment 2017
Date: 2013-11-09 18:26:27 +0000
From: @benbullard79

Confirmed. There are others, just from memory, nano, knemo, vuze packages, eclipse-swt, bouncycastle, some others maybe like kde-rootactions-servicemenu.

## Comment 2040
Date: 2013-11-10 19:15:54 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

I made a mistake.

unrar-4.20-3 also gives a "Medium without key".

## Comment 2097
Date: 2013-11-12 10:01:50 +0000
From: @itchka

I can find neither of these packages in the repositorys on my system. Is the bug set to the correct architecture? Please check and report your repository url's.

## Comment 2099
Date: 2013-11-12 13:39:30 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

$ urpmq --list-url
main (distrib5) http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/x86_64/media/main/release
main updates (distrib6) http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/x86_64/media/main/updates
contrib (distrib7) http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/x86_64/media/contrib/release
contrib updates (distrib8) http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/x86_64/media/contrib/updates
non-free (distrib9) http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/x86_64/media/non-free/release
non-free updates (distrib10) http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/x86_64/media/non-free/updates
restricted (distrib11) http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/x86_64/media/restricted/release
restricted updates (distrib12) http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/x86_64/media/restricted/updates
main32 http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/i586/media/main/release
main32 updates http://abf.rosalinux.ru/downloads/openmandriva2013.0/repository/i586/media/main/updates

For example, unrar is in non-free (distrib9). I just re-checked 2 minutes ago. Still says:

------
unrar-4.20-3-omv2013.0.x86_64.rpm: Medium without key (OK (RSA/SHA1, lun 09 sep 2013 11:46:59 ART, Key ID 180922d8bf81de15))
------

Note: "lun" is for "lunes", Spanish of "monday". "sep" is for September (Septiembre in Spanish)

## Comment 2110
Date: 2013-11-13 09:04:26 +0000
From: @itchka

I have the same repos as you. Can you run urpmi.update -a and then check whether unrar and the likes are still there? This is may be a problem on my system.

urpmq --list-url 
main2 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/main/release
main updates2 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/main/release
restricted2 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/main/release
restricted updates2 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/main/release
non-free2 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/main/release
non-free updates2 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/main/release
contrib2 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/contrib/release
contrib updates2 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/contrib/release
main3 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/i586/media/main/release
main updates3 http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/i586/media/main/updates

## Comment 2127
Date: 2013-11-13 19:53:22 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

# urpmi.update -a
el soporte «main (distrib5)» está al día
el soporte «main updates (distrib6)» está al día
el soporte «contrib (distrib7)» está al día
el soporte «contrib updates (distrib8)» está al día
el soporte «non-free (distrib9)» está al día
el soporte «non-free updates (distrib10)» está al día
el soporte «restricted (distrib11)» está al día
el soporte «restricted updates (distrib12)» está al día
el soporte «main32» está al día
el soporte «main32 updates» está al día

# rpm -qa | grep unrar
unrar-4.20-3-omv2013.0.x86_64

Checking in MCC, Software Management, I can see unrar. But it's pretty obvious to me, as unrar is already installed.

I went to http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/media/non-free/release and unrar exists, with the same version (4.20-3)

## Comment 2128
Date: 2013-11-13 19:55:28 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

Colin:

Uninstall unrar and then reinstall it. That's the way I re-checked (see comment 4).

Maybe you have additional valid keys loaded that I haven't.

## Comment 2132
Date: 2013-11-14 10:19:44 +0000
From: @itchka

I simply do not have any notion of the unrar file it is nether installed nor available through my urpmi/rpm setup. The only thing I can think of that might cause this difference is that I had occasion to rebuild my rpm database due to a lock up. Maybe when you rebuild the db it rejects all the packages with unmatched keys.
I am about to do some new installs for RC2 testing I will check for unrar on one of these.

## Comment 2137
Date: 2013-11-14 12:14:47 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

You can try:

$ unrar

If unrar is installed, it will print the help for the command.

Or:

$ ls /usr/bin/un*

If unrar is installed it will appears in the listing of the directory /usr/bin.

## Comment 2143
Date: 2013-11-14 17:31:52 +0000
From: @benbullard79

In a fresh install of 2013 Lx RC2 (x86_64) today and fully updated I got this:

# urpmi -v unrar
...
The following package has bad signature:
/var/cache/urpmi/rpms/unrar-4.20-3-omv2013.0.x86_64.rpm: Medium without key (OK (RSA/SHA1, Mon 09 Sep 2013 09:46:59 AM CDT, Key ID 180922d8bf81de15))
Do you want to continue installation ? (y/N)

Same for all the packages listed in comment # 1.

## Comment 2160
Date: 2013-11-15 13:33:23 +0000
From: @tpgxyz

Maybe packages from restricted repo are not signed at all ?

CC'ing Bero.

## Comment 2165
Date: 2013-11-15 15:22:18 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

(In reply to comment #11)
> Maybe packages from restricted repo are not signed at all ?
> 
> CC'ing Bero.

TPG:

unrar is in non-free.

Packages named by Ben:
eclipse-swt and bouncycastle are in main.
nano, knemo and vuze are in contrib.

Note that eclipse-swt is marked as "mdv2013", not "omv2013" (bug #54, which is marked as SOLVED !!!!)

I can understand that packages like flash (from Adobe) or skype (now Microsoft) be unsigned, as they are downloaded from the site of the company that produce them. But Ben named a lot of packages, even "KDE officials" like knemo, that it's supposed that must be signed.

## Comment 2167
Date: 2013-11-15 15:52:49 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

Another packages with invalid keys:

kde4-style-polyester-2.0.0-2-omv2013.0-x86_64.rpm: ID de clave no válida (OK (DSA/SHA1, dom 21 abr 2013 01:20:07 ART, Key ID 94e84b22d2aa6cb1))

kde4-style-crystal-2.0.5-1-omv2013.0-x86_64.rpm: ID de clave no válida (OK (DSA/SHA1, sáb 20 abr 2013 02:45:01 ART, Key ID d210639003984595))

kde4-style-nitrogen-3.3.3-2-omv2013.0-x86_64.rpm: ID de clave no válida (OK (DSA/SHA1, sáb 20 abr 2013 00:47:37 ART, Key ID cab980145f766f49))

kde4-style-skulpture-0.2.3-2-omv2013.0-x86_64.rpm: ID de clave no válida (OK (DSA/SHA1, sáb 20 abr 2013 18:03:47 ART, Key ID 6d07180b3b7939c1))

All of them are in contrib.

## Comment 2206
Date: 2013-11-17 15:54:05 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

More packages to review (they give "invalid key"):

lib64ts0.0_0-1.0-7-omv2013.0.x86_64.rpm
tslib-common-1.0-7-omv2013.0.x86_64.rpm

## Comment 2370
Date: 2013-11-22 12:02:13 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

I just opened bug #344 for the same problem, but in the Final Release Candidate version.

## Comment 2935
Date: 2014-01-10 09:40:15 +0000
From: @tpgxyz

This is a know bug of abf.io
Currenlty only new packages are signed, and we are working on mass resigning all packges for all repos

## Comment 3305
Date: 2014-02-25 09:25:22 +0000
From: @tpgxyz

Duplicate of bug #344

TPG
Cooker Team

*** This bug has been marked as a duplicate of bug #344 ***
